misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -94,4 +95,44 @@ fun Aead.decrypt(ciphertext: ByteString, aad: ByteArray? = null): ByteString {
   val decrypted = decryptedBytes.toByteString()
   decryptedBytes.fill(0)
   return decrypted
+}
+
+/**
+ * Extension function for convenient encryption of [ByteString]s.
+ * This function also makes sure that no extra copies of the plaintext data are kept in memory.
+ */
+fun DeterministicAead.encryptDeterministically(plaintext: ByteString, aad: ByteArray? = null
```

Comment:
is `Deterministically` redundant given it's a `DeterministicAead`?

Reply:
I wanted this to match the method name provided by the `DeterministicAead` interface for compatibility.
This way, it's more like method overloading rather then an extension

Reply:
This is an extension for Tink's `DeterministicAead` which uses the methods `encryptDeterministically`/`decryptDeterministically`, and this just wraps them. I believe Tink chose the naming scheme to ensure AEAD and DeterministicAead were not swappable.

Reply:
(woops, submitted at the same time)